src/Data/Unit/Polymorphic/Properties.agda

```diff
@@ -11,91 +11,78 @@ module Data.Unit.Polymorphic.Properties where
 
 open import Level
 open import Data.Sum using (inj₁)
-open import Data.Unit.Polymorphic.Base using (⊤; tt; _≤_)
+open import Data.Unit.Polymorphic.Base using (⊤; tt)
 open import Relation.Nullary
 open import Relation.Binary
 open import Relation.Binary.PropositionalEquality
 
 private
   variable
-    ℓ ℓ′ a b e : Level
+    ℓ : Level
 
 ------------------------------------------------------------------------
--- Decidable Equality and Ordering
+-- Equality
+------------------------------------------------------------------------
 
 infix 4 _≟_
 
-_≟_ : {ℓ : Level} → Decidable {A = ⊤ {ℓ}} _≡_
+_≟_ : Decidable {A = ⊤ {ℓ}} _≡_
 _ ≟ _ = yes refl
 
-infix 4 _≤?_
-
-_≤?_ : {a b ℓ : Level} → Decidable {a} {_} {b} {ℓ = ℓ}  _≤_
-_ ≤? _ = yes _
-
-------------------------------------------------------------------------
--- Setoid
-
 ≡-setoid : Setoid ℓ ℓ
 ≡-setoid = setoid ⊤
 
 ≡-decSetoid : DecSetoid ℓ ℓ
 ≡-decSetoid = decSetoid _≟_
 
 ------------------------------------------------------------------------
--- _≤_
-
--- Relational properties
-
-≤-reflexive : _≡_ {a} ⇒ (_≤_ {e = e})
-≤-reflexive _ = _
-
-≤-trans : Transitive (_≤_ {ℓ} {e = e})
-≤-trans _ _ = _
-
-≤-antisym : Antisymmetric _≡_ (_≤_ {ℓ} {e = e})
-≤-antisym  _ _ = refl
+-- Ordering
+------------------------------------------------------------------------
 
-≤-total : Total (_≤_ {ℓ} {e = e})
-≤-total _ _ = inj₁ _
+≡-total : Total {A = ⊤ {ℓ}} _≡_
+≡-total _ _ = inj₁ refl
 
+≡-antisym : Antisymmetric {A = ⊤ {ℓ}} _≡_ _≡_
+≡-antisym p _ = p
 
+------------------------------------------------------------------------
 -- Structures
 
-≤-isPreorder : IsPreorder _≡_ (_≤_ {ℓ} {e = e})
-≤-isPreorder = record
+≡-isPreorder : IsPreorder {ℓ} {_} {⊤ {ℓ}} _≡_ _≡_
```

Comment:
It occurs to me that for all the Structures and Bundles we probably want the level to be given explicitly rather than implicitly. It's very rare that structures or bundles are used in a context where Agda can infer the levels...

Reply:
The 'problem' here is with `IsPreorder` , right? It should have 2 explicit `Level` parameters?

Reply:
Yes, as should `≤-preorder`.

Reply:
But that is independent of this PR, right?  Can't we pull this in as-is, and then make an issue to 'improve' `IsPreorder` and `≤-preorder` appropriately?

Reply:
Yup, fair enough! Merging in.